test/e2e/util/hypershift_framework.go

```diff
@@ -322,3 +327,18 @@ func (h *hypershiftTest) teardownHostedCluster(ctx context.Context, hc *hyperv1.
 		}
 	}
 }
+
+func (h *hypershiftTest) summarizeHCConditions(hostedCluster *hyperv1.HostedCluster) {
+	conditions := hostedCluster.Status.Conditions
+
+	if conditions != nil {
+		h.Logf("Summarizing false conditions in the HostedCluster")
+		for _, condition := range conditions {
+			if condition.Status == "False" {
+				h.Logf("Condition %s is false for HostedCluster %s: %s", condition.Type, hostedCluster.Name, condition.Message)
```

Comment:
The HostedCluster never changes, so let's log it on L335, once, and only log the unique part of each condition here. Similarly, we're only logging false conditions, so  Also, let's emit the reason with the message, if present. 

```go
for _, condition := range conditions {
    if condition.Status == metav1.ConditionFalse {
        msg := fmt.Sprintf("%s: %s", condition.Type, condition.Reason)
        if condition.Message != "" {
            msg += fmt.Sprintf": %s", condition.Message)
        }
    }
}
```